lib/task.js

```diff
@@ -440,40 +565,82 @@ export class Task {
   }
 
   /**
-  Determine when the right time is to issue another reclaim then schedule it
-  via set timeout.
-  */
-  scheduleReclaim(claim) {
-    // Figure out when to issue the next claim...
-    let takenUntil = (new Date(claim.takenUntil) - new Date());
-    let nextClaim = takenUntil / this.runtime.task.reclaimDivisor;
+   * Resolves all of the non-primary claims for this task, optionally
+   * adding an artifact to each one named "public/superseded-by"
+   * containing the taskId/runId of the primary claim.
+   *
+   * @param {String} primaryTaskId taskId of the primary task
+   * @param {Integer} primaryRunId runId of the primary task
+   * @param {Boolean} addArtifacts if true, add the superseded-by artifacts
+   * @param {String} reason the exception reason passed to the queue
+   */
 
-    // This is tricky ensure we have logs...
-    this.runtime.log('next claim', {
-      taskId: this.status.taskId,
-      runId: this.runId,
-      time: nextClaim
-    });
+  async resolveSuperseded(primaryTaskId, primaryRunId, addArtifacts, reason) {
+    let queue = this.runtime.queue;
+    let supersedes = [];
 
-    // Figure out when we need to make the next claim...
-    this.clearClaimTimeout();
+    try {
+      await Promise.all(this.claims.map(async function(c){
+        let taskId = c.status.taskId;
+        let runId = c.runId;
+        if (taskId == primaryTaskId && runId == primaryRunId) {
+          return;
+        }
 
-    this.claimTimeoutId =
-      setTimeout(function() {
-        async () => {
-          await this.reclaimTask();
-        }()
-      }.bind(this), nextClaim);
+        await queue.reportException(taskid, runId, {reason});
+
+        if (addArtifacts) {
+          let task = c.task;
+          // set the artifact expiration to match the task
+          let expiration = task.expires || taskcluster.fromNow(task.deadline, "1 year");
+          expiration = new Date(Math.min(expiration, new Date(task.expires)));
```

Comment:
I troubleshooted this a little bit, and it looks like this results in "invalidate date" because expiration here is a string, and when calling Math.min() involving a string results in NaN.  pass that to Date() and things are not so good..

> new Date(Math.min("2016-12-22T18:04:45.420Z", new Date()))
Invalid Date